Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -302,20 +308,25 @@ private class TopSitesLayout: UICollectionViewLayout {
 
     override init() {
         super.init()
-        setupForOrientation(UIApplication.sharedApplication().statusBarOrientation)
+        let rootWindow = (UIApplication.sharedApplication().delegate as! AppDelegate).window!
+        setupForTraitCollection(rootWindow.traitCollection, viewSize: rootWindow.bounds.size)
```

Comment:
Hm, having a `UIView` reach into `UIApplication` like this feels like a red flag. Do you think it would make sense to turn `thumbnailCols` into a computed property that can dynamically calculate the number of thumbnails from the view's current width?